File: apps/api/AssetManagement.Api/Models/ApplicationType.cs
namespace AssetManagement.Api.Models;

public class ApplicationType
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public bool IsArchived { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Application> Applications { get; set; } = [];
    public ICollection<CustomFieldDefinition> CustomFieldDefinitions { get; set; } = [];
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/ApplicationType.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.time.Instant
import java.util.*

@Entity
@Table(name = "application_types")
class ApplicationType(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    override var id: UUID = UUID.randomUUID(),

    @Column(name = "name", nullable = false)
    override var name: String = "",

    @Column(name = "description")
    var description: String? = null,

    @Column(name = "is_archived", nullable = false)
    override var isArchived: Boolean = false,

    @org.hibernate.annotations.CreationTimestamp
    @Column(name = "created_at", nullable = false)
    var createdAt: Instant = Instant.now(),

    @org.hibernate.annotations.UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    override var updatedAt: Instant = Instant.now(),

    @OneToMany(mappedBy = "applicationType")
    var applications: MutableList<Application> = mutableListOf(),

    @OneToMany(mappedBy = "applicationType", cascade = [CascadeType.ALL], orphanRemoval = true)
    override var customFieldDefinitions: MutableList<CustomFieldDefinition> = mutableListOf()
) : ArchivableType
